### ISSUE TITLE
Failing to write out string arrays

### ISSUE THREAD

**Issue by bennahugo** [casacore/python-casacore#174]
Hi I'm trying to fill out the meta data for a casacore gain table with python 3 and casacore 3.0. 

However, I'm running into a little snag where casacore does not accept string arrays:
```
            with tbl("%s::ANTENNA" % str(filename), ack=False, readonly=False) as t:
                t.addrows(nrows=len(db.anttype))
                t.putcol("OFFSET", db.antoffset[antorder])
                t.putcol("POSITION", db.antpos[antorder])
                t.putcol("TYPE", np.array(list(map(str, db.anttype)), dtype=str)[antorder])
                t.putcol("DISH_DIAMETER", db.antdishdiam[antorder])
                t.putcol("FLAG_ROW", db.antflagrow[antorder])
                t.putcol("MOUNT", np.array(list(map(str, db.antmount)), dtype=str)[antorder])
                t.putcol("NAME", np.array(list(map(str, db.antnames)), dtype=str)[antorder])
                t.putcol("STATION", np.array(list(map(str, db.antstation)), dtype=str)[antorder])
```
with the following error
```
 - 20:29:55 - main               [io] [0.1/0.6 0.9/1.4 0.8Gb] I/O handler for load None save -1 failed with exception: PycArray: unknown python array data type
 - 20:29:55 - main               [io] [0.1/0.6 0.9/1.4 0.8Gb] Traceback (most recent call last):
  File "/home/bhugo/workspace/CubiCal/cubical/workers.py", line 445, in _io_handler
    solver.gm_factory.close()
  File "/home/bhugo/workspace/CubiCal/cubical/machines/abstract_machine.py", line 862, in close
    db.close()
  File "/home/bhugo/workspace/CubiCal/cubical/database/casa_db_adaptor.py", line 467, in close
    self.__export()
  File "/home/bhugo/workspace/CubiCal/cubical/database/casa_db_adaptor.py", line 452, in __export
    casa_caltable_factory.create_B_table(self, "G:gain")
  File "/home/bhugo/workspace/CubiCal/cubical/database/casa_db_adaptor.py", line 242, in create_B_table
    viscal_label="B Jones" if diag else "D Jones")
  File "/home/bhugo/workspace/CubiCal/cubical/database/casa_db_adaptor.py", line 60, in init_empty
    t.putcol("TYPE", np.array(list(map(str, db.anttype)), dtype=str)[antorder])
  File "/home/bhugo/workspace/venv3/lib/python3.6/site-packages/casacore/tables/table.py", line 1157, in putcol
    self._putcol(columnname, startrow, nrow, rowincr, value)
RuntimeError: PycArray: unknown python array data type
```

The same code works perfectly fine under python 2.7

**Comment by bennahugo** [comment]
@gijzelaerr any ideas on how to work around the problem?

**Comment by gijzelaerr** [comment]
update to a newer casacore. 

I'm trying to get casacore 3.1.1 and python-casacore 3.1.0 out the door so i can roll a binary wheel for both py2 and py3, that would make it easier to upgrade.

**Comment by gijzelaerr** [comment]
This issue has been fixed in casacore 3.1+
